x-pack/filebeat/input/http_endpoint/validate.go

```
@@ -50,5 +63,37 @@ func (v *apiValidator) ValidateHeader(r *http.Request) (int, error) {
 		return http.StatusUnsupportedMediaType, fmt.Errorf("Wrong Content-Type header, expecting %v", v.contentType)
 	}
 
+	if v.hmacHeader != "" && v.hmacKey != "" && v.hmacType != "" {
+		// We need access to the request body to validate the signature.
+		buf, _ := ioutil.ReadAll(r.Body)
```

This should return an error.